test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -212,34 +212,197 @@ func (s *snapshottableTestSuite) DefineTests(driver TestDriver, pattern testpatt
 	})
 }
 
 // WaitForSnapshotReady waits for a VolumeSnapshot to be ready to use or until timeout occurs, whichever comes first.
-func WaitForSnapshotReady(c dynamic.Interface, ns string, snapshotName string, Poll, timeout time.Duration) error {
+func WaitForSnapshotReady(c dynamic.Interface, ns string, snapshotName string, poll, timeout time.Duration) error {
 	framework.Logf("Waiting up to %v for VolumeSnapshot %s to become ready", timeout, snapshotName)
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
+
+	if successful := WaitUntil(poll, timeout, func() bool {
 		snapshot, err := c.Resource(SnapshotGVR).Namespace(ns).Get(context.TODO(), snapshotName, metav1.GetOptions{})
 		if err != nil {
-			framework.Logf("Failed to get claim %q, retrying in %v. Error: %v", snapshotName, Poll, err)
-			continue
-		} else {
-			status := snapshot.Object["status"]
-			if status == nil {
-				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
-				continue
-			}
-			value := status.(map[string]interface{})
-			if value["readyToUse"] == true {
-				framework.Logf("VolumeSnapshot %s found and is ready after %v", snapshotName, time.Since(start))
-				return nil
-			} else if value["ready"] == true {
-				framework.Logf("VolumeSnapshot %s found and is ready after %v", snapshotName, time.Since(start))
-				return nil
-			} else {
-				framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
-			}
+			framework.Logf("Failed to get claim %q, retrying in %v. Error: %v", snapshotName, poll, err)
+			return false
 		}
+
+		status := snapshot.Object["status"]
+		if status == nil {
+			framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
+			return false
+		}
+		value := status.(map[string]interface{})
+		if value["readyToUse"] == true || value["ready"] == true {
+			framework.Logf("VolumeSnapshot %s found and is ready", snapshotName)
+			return true
+		}
+
+		framework.Logf("VolumeSnapshot %s found but is not ready.", snapshotName)
+		return false
+	}); successful {
+		return nil
 	}
+
 	return fmt.Errorf("VolumeSnapshot %s is not ready within %v", snapshotName, timeout)
 }
+
+// WaitForSnapshotContentDelete waits until a snapshot has been deleted
+func WaitForSnapshotContentDelete(c dynamic.Interface, ns string, snapshotContentName string, poll, timeout time.Duration) error {
+	framework.Logf("Waiting up to %v for VolumeSnapshotContent %s to be deleted", timeout, snapshotContentName)
+
+	if successful := WaitUntil(poll, timeout, func() bool {
+		_, err := c.Resource(SnapshotContentGVR).Namespace(ns).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+		if err != nil && apierrors.IsNotFound(err) {
+			framework.Logf("VolumeSnapshotContent %v is not found and has been deleted", snapshotContentName)
+			return true
+		} else if err != nil {
+			framework.Logf("Get VolumeSnapshotContent %v returned an unrelated error: %v", snapshotContentName, err.Error())
```

Comment:
What is an unrelated error?

Reply:
Basically, any error that isn't NotFound. I just now removed the word "unrelated".